evm_arithmetization/src/cpu/kernel/interpreter.rs

```diff
@@ -141,12 +141,11 @@ pub(crate) fn simulate_cpu_and_get_user_jumps<F: Field>(
 pub(crate) fn generate_segment<F: Field>(
     max_cpu_len: usize,
     index: usize,
-    inputs: &GenerationInputs,
+    inputs: GenerationInputs,
 ) -> anyhow::Result<(RegistersState, RegistersState, MemoryState)> {
     let init_label = KERNEL.global_labels["init"];
     let initial_registers = RegistersState::new_with_main_label();
-    let mut interpreter =
-        Interpreter::<F>::new_with_generation_inputs(init_label, vec![], inputs.clone());
```

Comment:
@Nashtare The saved clone is here.

Reply:
But you still had to clone inputs before calling `generate_segment` (when you do `let inputs2 = inputs.clone()`)